libsolidity/ast/Types.h

```diff
@@ -54,6 +54,7 @@ using TypePointers = std::vector<Type const*>;
 using rational = boost::rational<bigint>;
 using TypeResult = util::Result<Type const*>;
 using BoolResult = util::Result<bool>;
+using FunctionDefinitionResult = util::Result<FunctionDefinition const*>;
```

Comment:
I'm not sure if this is so useful to have in the header - maybe an explicit type would be better readable?